apps/app_rand.c

```diff
@@ -84,7 +85,8 @@ int opt_rand(int opt)
         return loadfiles(opt_arg());
         break;
     case OPT_R_WRITERAND:
-        save_rand_file = opt_arg();
+        OPENSSL_free(save_rand_file);
```

Comment:
*If* there is situation when OPENSSL_free would be required here, then it would be appropriate to `save_rand_file = NULL` after line 69. Otherwise we would be looking at double-free. No?